src/Channels.Networking.Windows.RIO/Internal/RioThread.cs

```diff
@@ -120,7 +126,10 @@ private static void OnThreadStart(object state)
                             RioTcpConnection connection;
                             if (thread._connections.TryGetValue(result.ConnectionCorrelation, out connection))
                             {
-                                connection.Complete(result.Status, result.RequestCorrelation, result.BytesTransferred);
+                                ThreadPool.QueueUserWorkItem(
```

Comment:
I was trying to avoid dispatching as long as possible. I'm not convinced this is needed yet. RIO was doing 20K RPS on my box and some changes took it down to 4K RPS so this is definitely unrelated.